### PR TITLE
Aarav/donation form stuff clean

### PR DESCRIPTION
**Pull request by constrictingsnake**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [ ] I have performed a self-review and reviewed the changed files
- [ ] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [ ] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Summary by CodeRabbit

* **New Features**
  * Added "Donation Form" editor in the control panel for customizing donation content.
  * Upload banner images and edit form content with formatting tools (bold, italic, headings, lists).
  * Full-screen preview modal to see how the form appears to donors before saving.
  * Donation form now displays customized banner and formatted content to donors during checkout.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Hack4Impact-UMD/journey-home/pull/110?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: app/control-panel/donation-form/page.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                        className={`h-80 overflow-y-auto px-4 pb-2 [&_.ProseMirror]:outline-none [&_.ProseMirror:focus]:outline-none${isEditing ? " cursor-text" : ""}`}
+                        onClick={() => isEditing && editor?.commands.focus()}
+                    >
+                        <EditorContent editor={editor} className={isEditing ? "pt-4" : "hidden"} />
+                        {!isEditing && (
+                            <div
+                                className="prose max-w-none pt-4"
+                                dangerouslySetInnerHTML={{ __html: formData?.content ?? "" }}
+                            />
+                        )}
```
[the remark this answers is not in the document]
Comment (constrictingsnake) [reply app/control-panel/donation-form/page.tsx:186]: only admins can modify this, and if an attacker has access to an admin account then the donation form text xss injection is the least of the concerns
